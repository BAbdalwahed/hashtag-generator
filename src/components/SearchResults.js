import React from "react";
import PropTypes from "prop-types";

const SearchResults = ({ searchResults }) => {
  if (!searchResults) {
    return (
      <p className="no-results-message">
        Sorry, we couldn&apos;t find anything.
      </p>
    );
  }
  return <div className="search-results">{searchResults}</div>;
};

SearchResults.propTypes = {
  searchResults: PropTypes.string.isRequired,
};

export default SearchResults;
