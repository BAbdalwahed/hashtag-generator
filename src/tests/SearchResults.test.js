import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  let results = "test test test test";
  xit("renders search results correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <SearchResults searchResults={results} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  xit("renders no search results correctly", () => {
    results = "";
    const { asFragment } = render(
      <BrowserRouter>
        <SearchResults searchResults={results} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
