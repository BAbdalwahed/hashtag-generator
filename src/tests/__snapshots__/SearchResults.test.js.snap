// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders no search results correctly 1`] = `
<DocumentFragment>
  <p
    class="no-results-message"
  >
    Sorry, we couldn't find anything.
  </p>
</DocumentFragment>
`;

exports[`Search Results renders search results correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    test test test test
  </div>
</DocumentFragment>
`;
